crates/bevy_reflect/src/impls/std.rs

```diff
@@ -662,29 +711,48 @@ impl<T: Reflect, const N: usize> Typed for [T; N] {
         CELL.get_or_insert::<Self, _>(|| TypeInfo::Array(ArrayInfo::new::<Self, T>(N)))
     }
 }
-
-// TODO:
-// `FromType::from_type` requires `Deserialize<'de>` to be implemented for `T`.
-// Currently serde only supports `Deserialize<'de>` for arrays up to size 32.
-// This can be changed to use const generics once serde utilizes const generics for arrays.
-// Tracking issue: https://github.com/serde-rs/serde/issues/1937
-macro_rules! impl_array_get_type_registration {
-    ($($N:expr)+) => {
-        $(
-            impl<T: Reflect > GetTypeRegistration for [T; $N] {
-                fn get_type_registration() -> TypeRegistration {
-                    TypeRegistration::of::<[T; $N]>()
-                }
-            }
-        )+
-    };
+impl<T: Reflect, const N: usize> GetTypeRegistration for [T; N] {
+    fn get_type_registration() -> TypeRegistration {
+        TypeRegistration::of::<[T; N]>()
+    }
 }
 
-impl_array_get_type_registration! {
-     0  1  2  3  4  5  6  7  8  9
-    10 11 12 13 14 15 16 17 18 19
-    20 21 22 23 24 25 26 27 28 29
-    30 31 32
```

Comment:
This could also potentially be split out into a separate PR if we wanted to to reduce the noise of this one (not super necessary though as I don't expect this to be a contentious change).